File: docs/WidgetMill/packages/spec/src/nodes/padding.ts
import { z } from "zod";
import { EdgeInsets } from "../complex/edgeInsets";
import { NodeRef } from "../node-ref";

/** Padding — spec §2.10. `padding` é obrigatório. */
export const PaddingNode = z.object({
  type: z.literal("padding"),
  props: z.object({ padding: EdgeInsets }),
  child: NodeRef.optional(),
});

File: docs/WidgetMill/flutter/sdui_flutter/lib/src/builders/padding.dart
import 'package:flutter/widgets.dart';
import '../model/sdui_node.dart';
import '../parsing/parsers.dart';
import '../renderer.dart';

/// Padding — spec §2.10.
Widget buildPadding(BuildContext context, SduiNode node, SduiRenderer r) {
  final p = node.props;
  return Padding(
    padding: parseEdgeInsets(p['padding']) ?? EdgeInsets.zero,
    child: r.maybeRender(context, node.child),
  );
}
